### PR TITLE
Store Helm charts in Google Artifact Registry

### PR DESCRIPTION
**Pull request by counik**
This PR updates adds a Github Action to push new Helm charts to Artifact Registry. It's currently configured to run anytime a commit is merged to `main` that modifies `Chart.yaml`. It can also be run manually.

I've simplified how we authenticate between Github and Google, because I learned that how I'd done it previously was redundant - there's no need to pull down and install the entire `gcloud` CLI just to setup a connection between Helm and Artifact Registry.

Aside from getting rid of several unnecessary dependencies, this cut the run time from ~30 seconds down to ~15 seconds.

Additionally, we no longer need the `index.yaml` file, or the command that re-generates it: https://helm.sh/docs/topics/registries/#migrating-from-chart-repos

### CODE REVIEW

File: package-chart.sh [hunk 1]
```diff
@@ -1,4 +1,3 @@
 #!/usr/bin/env bash
 
 helm package basic-web-service -d charts
-helm repo index charts
```
Comment (counik) [inline package-chart.sh:4]: `index.yaml` is not used by OCI artifact registries.

File: .github/workflows/chart_push.yml [hunk 2]
```diff
@@ -1,23 +1,37 @@
...
+        GAR_KEY_BASE64: ${{ secrets.GAR_KEY_BASE64 }}
+        REGISTRY_URL_HTTP: "https://${{ env.REGISTRY_HOSTNAME }}"
       run: |
-        gcloud auth application-default print-access-token | helm registry login -u oauth2accesstoken \
-        --password-stdin https://us-central1-docker.pkg.dev
+        echo $GAR_KEY_BASE64 | helm registry login -u _json_key_base64 --password-stdin \
+        $REGISTRY_URL_HTTP
     - name: 'Upload chart'
+      env:
+        REGISTRY_URL_OCI: "oci://${{env.REGISTRY_HOSTNAME}}/uwit-mci-iam/helm-charts-test"
```
Comment (counik) [inline .github/workflows/chart_push.yml:34]: This will be changed after testing. Currently, it's pointed at a test registry.

**Comment by counik** [comment]
1. If the chart has already been uploaded it overwrites the old one (sort of). My understanding is that this is actually defined somewhere in the OCI spec, but I haven't looked too deeply in it. Basically it shifts the tag "pointer" to point to the newly uploaded image, and leaves the previous image intact, but without the tag.

Note that this only applies if you have generated a _new_ image archive with the same image tag (aka it has a different digest hash). Uploading the exact same archived chart is an idempotent operation.

Here's what this looks like in Artifact Registry
Screenshot 2023-07-25 at 9 53 37 AM

2. I'd guess it does? I haven't tested though. I thought about extending this to more easily support new charts, but decided that since we hadn't added any new ones in a year, I would keep this a bit simpler.

3. I'm fine either way. This was also an intentional decision (and not just a lazy one, I swear). I thought that this would safeguard against people forgetting to bump the version number in `Chart.yaml`, since they would need to do the added manual step of generating and committing the tar file.

**Comment by counik** [comment]
@miker985 

New commit checks if the chart was already uploaded to the registry, and throws an error if it already exists. Also added a couple error annotations to make failures more helpful.

File: .github/workflows/chart_push.yml [hunk 3]
```diff
@@ -1,23 +1,47 @@
...
+        echo $GAR_KEY_BASE64 | helm registry login -u _json_key_base64 --password-stdin \
+        $REGISTRY_URL_HTTP
+    - name: 'Check if chart version already exists in registry'
+      env:
+        CHART_URL_OCI: "oci://${{ env.REGISTRY_HOSTNAME }}/uwit-mci-iam/helm-charts-test/basic-web-service"
+        ERROR_CHART_EXISTS: "::error:: basic-web-service:${{ env.CHART_VERSION }} already exists in registry, did you forget to bump the version in Chart.yaml?"
       run: |
-        gcloud auth application-default print-access-token | helm registry login -u oauth2accesstoken \
-        --password-stdin https://us-central1-docker.pkg.dev
+        helm pull $CHART_URL_OCI --version $CHART_VERSION && echo $ERROR_CHART_EXISTS && exit 1 || echo 'Uploading new chart'
```
Comment (miker985) [inline .github/workflows/chart_push.yml:42]: I'm always really bad at these sorts of statements in bash and encourage wrapping the compound statements in `(` `)` symbols for clarity.

**Review by miker985 — APPROVED** [review]

**Pull request merged**